Neos.Media/Classes/Domain/Model/ThumbnailGenerator/ImageThumbnailGenerator.php

```diff
@@ -80,4 +81,14 @@ public function refresh(Thumbnail $thumbnail)
             throw new Exception\NoThumbnailAvailableException($message, 1433109654, $exception);
         }
     }
+
+    /**
+     * @param PersistentResource $resource
+     * @param array $adjustments
+     * @return array
+     */
+    protected function processResource(PersistentResource $resource, array $adjustments)
+    {
+        return $this->imageService->processImage($resource, $adjustments);
+    }
```

Comment:
@aertmann It can be easily overriden with custom generators.
This way we could for example use a custom generator instead of the AOP in the ImageOptimizer package.